### PR TITLE
Adding presto data source

### LINKED ISSUE #1689: Add Presto as a supported datasource :mag: :electric_plug:
We need to increase the number of datasources that MindsDB supports. This task should add a new datasource for connecting to Presto. You can get more info on [Presto](https://prestodb.io/docs/current/) docs. 
If possible, please include a test for the datasource. You can check the example [here](https://github.com/mindsdb/datasources/tree/staging/tests).

>Note: if you are familiar with another datasource that MindsDB doesn't support, feel free to work on that and make a PR.

## Steps :male_detective: :female_detective: 

- [ ] Frok datasources repo
- [ ] Add new datasource file in https://github.com/mindsdb/datasources/tree/staging/mindsdb_datasources/datasources
- [ ] Add test(optional)
- [ ] Make a PR to staging branch

## Additional rewards :1st_place_medal: 

Each datasource PR brings  :four:  point for entry into the draw for a :computer:  Deep Learning Laptop powered by the NVIDIA RTX 3080 Max-Q GPU or other swag :shirt: :bear: .  For more info check out https://mindsdb.com/hacktoberfest/

### PR DESCRIPTION
**Pull request by MichaelLantz**
Fixes mindsdb/mindsdb#1689

## Please describe what changes you made, in as much detail as possible
  - Adding datasource for Presto
  - Adding test against Presto datasource
  - Adding initialization check for Presto datasource
  - Updating dependencies / requirements for new presto python library

datasources

### CODE REVIEW

**Comment by MichaelLantz** [comment]
Still attempting to get my local tests working since all but file data sources test fail but the core of the data source has been created.

File: mindsdb_datasources/__init__.py [hunk 1]
```diff
@@ -69,3 +69,9 @@
 except ImportError:
     print("Cassandra Datasource is not available by default. If you wish to use it, please install mindsdb_native[cassandra]")
     CassandraDS = None
+
+try:
+    from mindsdb_datasources.datasources.presto_ds import PrestoDS
+except:
+    print("PrestoDS is not available by default. Please install snowflake's ODBC driver and pyodbc to use it!")
```
Comment (ZoranPandovski) [inline mindsdb_datasources/__init__.py]: Please change the snowflake message

File: mindsdb_datasources/datasources/presto_ds.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        con = prestodb.dbapi.connect(
+            schema=self.schema,
+            user=self.user,
+            password=self.password,
+            host=self.host,
+            port=self.port,
+            http_scheme=self.protocol
+        )
+
+        df = pd.read_sql(q, con=con)
```
Comment (ZoranPandovski) [inline mindsdb_datasources/datasources/presto_ds.py]: Why we need pandas here?

File: mindsdb_datasources/datasources/presto_ds.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        con.close()
```
Comment (ZoranPandovski) [inline mindsdb_datasources/datasources/presto_ds.py]: We need to close the connection last

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

**Comment by ZoranPandovski** [comment]
****CLA Assistant Lite bot**** All contributors have signed the CLA  ✍️

**Comment by MichaelLantz** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**